### PR TITLE
[SYCLomatic] Add test case for 2 CUB APIs

### PR DESCRIPTION
**Pull request by yihwang-nv**
Add feature test case for cub::DeviceReduce::Reduce and cub::DeviceReduce::ReduceByKey

Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: features/feature_case/cub/cub_device_reduce.cu [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    print_data<int>(&expect_ans, 1, true);
+    std::cout << "current result:\n";
+    print_data<int>(d_out, 1);
+    return false;
+  }
+  return true;
+}
+
+int main() {
+  if (test_device_reduce())
```
Comment (ShengchenJ) [inline features/feature_case/cub/cub_device_reduce.cu]: The case will always be true. Please return not zero value if the result is not expected.

File: features/feature_case/cub/cub_device_reduce_by_key.cu [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    std::cout << "current result:\n";
+    print_data<int>(d_agg, expect_num);
+    return false;
+  }
+
+  return true;
+}
+
+int main() {
+  if (test_device_reduce_by_key())
```
Comment (ShengchenJ) [inline features/feature_case/cub/cub_device_reduce_by_key.cu]: same as before.

**Review by ShengchenJ — APPROVED** [review]
LGTM

**Pull request merged**